app/src/frontend/building/data-components/like-data-entry.tsx

```diff
@@ -21,29 +29,31 @@ const LikeDataEntry: React.FunctionComponent<any> = (props) => { // TODO: remove
             </div>
             <p>
                 {
-                    (props.value != null)?
-                        (props.value === 1)?
-                            `${props.value} person likes this building`
-                            : `${props.value} people like this building`
+                    (props.totalLikes != null)?
+                        (props.totalLikes === 1)?
+                            `${props.totalLikes} person likes this building`
+                            : `${props.totalLikes} people like this building`
                         : "0 people like this building so far - you could be the first!"
                 }
             </p>
-            <input className="form-check-input" type="checkbox"
-                id="like" name="like"
-                checked={!!props.building_like}
-                disabled={props.mode === 'view'}
-                onChange={props.onLike}
-            />
-            <label htmlFor="like" className="form-check-label">
+            <label className="form-check-label">
+                <input className="form-check-input" type="checkbox"
+                    name="like"
+                    checked={!!props.userLike}
+                    disabled={props.mode === 'view'}
+                    onChange={e => props.onLike(e.target.checked)}
+                />
```

Comment:
the for/htmlFor attribute works correctly only when the referenced id is unique in the scope of the whole page (which was not the case) nesting input inside label is more reliable

Reply:
this is to make the label part of the click area for the checkbox (clicking the label toggles the checkbox) - which is advisable in general but especially on mobile